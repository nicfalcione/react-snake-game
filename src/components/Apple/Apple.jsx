import React from 'react';
import { useSelector } from 'react-redux';
import './Apple.css';

const Apple = () => {
    
    const { appleXPos, appleYPos } = useSelector((state) => ({
        appleXPos: state.apple.xPos,
        appleYPos: state.apple.yPos
    }));

    return (
        <div data-testid='apple' className='game-apple' style={{ top: appleYPos + 'px', left: appleXPos + 'px' }}></div>
    );
}

export default Apple;