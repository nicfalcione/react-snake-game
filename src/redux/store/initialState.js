import { RIGHT } from '../../constants/constants';

const defaultState = {
    isRunning: true,
    gameOver: false,
    direction: RIGHT,
    score: 0,
    speed: 100,
    snakeSegments:[
        {
            key: 1,
            xPos: 380,
            yPos: 220
        },
        {
            key: 2,
            xPos: 360,
            yPos: 220
        },
        {
            key: 3,
            xPos: 340,
            yPos: 220
        }
    ],
    apple: {
        xPos: 400,
        yPos: 400
    }
};

export default defaultState;