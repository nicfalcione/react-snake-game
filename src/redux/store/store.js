import { createStore } from '@reduxjs/toolkit';
import { gameStateReducer } from '../reducers/gameStateReducer';

const store = createStore(gameStateReducer);

export default store;