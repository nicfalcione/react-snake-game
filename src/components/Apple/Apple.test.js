import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import store from '../../redux/store/store';
import Apple from './Apple';
 
const setupRender = () => {
    return render(<Provider store={store}><Apple/></Provider>);
}

describe('Apple Component test', () => {
    it('Apple renders', () => {
        const { queryByTestId } = setupRender();
        expect(queryByTestId('apple')).toBeInTheDocument();
    })

    it('should match snapshot', () => {
        const { baseElement } = setupRender();
        expect(baseElement).toMatchSnapshot();
    })
})