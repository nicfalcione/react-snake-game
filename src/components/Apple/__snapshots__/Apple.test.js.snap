// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apple Component test should match snapshot 1`] = `
<body>
  <div>
    <div
      class="game-apple"
      data-testid="apple"
      style="top: 400px; left: 400px;"
    />
  </div>
</body>
`;
